source/view.c

```diff
@@ -1728,7 +1728,9 @@ static void rofi_view_trigger_global_action(KeyBindingAction action) {
       state->retv = MENU_CUSTOM_INPUT;
     }
 
-    state->quit = TRUE;
+    if (selected != UINT32_MAX) {
+      state->quit = TRUE;
+    }
```

Comment:
And checking the selected value here, before defining it's time to quit, would keep rofi open when `require_input` is enabled and there's no input. The user can still quit rofi normally by using the expected method, kb-cancel.